Enhance hammer_ping for 1.x

xiangce opened the pull request:
- now can handle some unexpected outputs
- update interfaces

bfahr commented on insights/parsers/hammer_ping.py:
```diff
@@ -1,21 +1,13 @@
+=============================================
 
 The hammer ping parser reads the output of ``hammer ping`` and turns it into
 a dictionary of service statuses in the ``data`` property.  Each service has
```
@xiangce the `data` attribute is mentioned here, but not shown in the examples or defined as an attribute.  Would you mind to add docstrings for the two attributes `data` and `error`?

xiangce replied:
@bfahr - Sure, I will modify the docstrings in next commit. Thanks.

xiangce commented on insights/parsers/hammer_ping.py:
```diff
@@ -1,21 +1,13 @@
 The hammer ping parser reads the output of ``hammer ping`` and turns it into
 a dictionary of service statuses in the ``data`` property.  Each service has
-its status stored in the 'status' key and its response in the 'repsonse' key;
+its status stored in the 'status' key and its response in the 'respsonse' key;
```
Thanks Paul

xiangce replied:
Your code looks good to me, I will use it directly.

xiangce commented on insights/parsers/hammer_ping.py:
```diff
@@ -31,68 +23,77 @@
+    def is_ok(self):
+        """bool: Are all services OK?"""
+        if not self.errors:
+            if self.data and all(svc['status'].lower() == 'ok' for svc in self.data.values()):
```
Make sense, will do it

xiangce commented:
@PaulWay - Would you please to check this PR again? If you think it's OK, please approve it then I can merge it. Thanks.

xiangce commented on insights/parsers/hammer_ping.py:
```diff
@@ -30,69 +22,75 @@
+        """list: The list of service names in order."""
+        self.errors = []
+        """list: Any error messages encountered during parsing."""
+        self.is_ok = False
```
@bfahr - OK, will do it in next commit.

xiangce commented on insights/parsers/hammer_ping.py:
```diff
@@ -30,69 +22,75 @@
+        self.errors = []
+        """list: Any error messages encountered during parsing."""
+        self.is_ok = False
+        """bool: Were all services read correctly and are they all in 'ok' state?"""
```
My fault, will do it. Thanks a lot.

xiangce commented:
@bfahr - I reverted the removed APIs and marked them as deprecated in docstring. Please check it again.

BTW, in order to mark the "self.data" individually, I added the "__init__" function.
